utils/src/main/scala/za/co/absa/enceladus/utils/validation/field/FieldValidatorDateTime.scala

```diff
@@ -0,0 +1,121 @@
+    val defaultTimeZone = field.getMetadata("timezone")
+    patternConversionIssues(pattern, defaultValue) ++
+      defaultTimeZoneIssues(defaultTimeZone) ++
+      patternAnalysisIssues(pattern, defaultValue, defaultTimeZone)
+  }
+
+  private def generateRow(rowType: String, timestamp: Timestamp)
+                         (implicit parser: EnceladusDateTimeParser): (String, String) = {
+    val q = "\""
+    val timestampAsString = parser.format(timestamp)
+    (s"Failed to convert $rowType ($q$timestampAsString$q) using ${parser.pattern.toString}", timestampAsString)
+  }
+
+  private def patternConversionIssues(pattern: DateTimePattern, defaultValue: Option[String]): Seq[ValidationIssue] = {
+
+    import FieldValidatorDateTime._
+    import spark.implicits._
+
+    val descriptionColName = "description"
+    val sourceColName = "source"
+    val convertedColName = "converted"
+    val q = "\""
+
+    try {
+      implicit val parser: EnceladusDateTimeParser = EnceladusDateTimeParser(pattern)
+      // Checking pattern's ability to be used in formatting date/time values
+
+      val data = Seq(
+        generateRow("current time", exampleDate),
+        generateRow("epoch", epochStart),
+        generateRow("epoch day end", epochStartDayEnd)
+      ) ++ defaultValue.map(
+        s =>(s"Default value $q$s$q does not adhere to ${pattern.toString}", s)
+      ).toSeq
+      val df =  data.toDF(descriptionColName, sourceColName)
+      val converted = conversion(col(sourceColName), pattern)
+
+      val dfWithConversion = df.select(
+        col(descriptionColName),
+        col(sourceColName),
+        converted.as(convertedColName),
+        converted.isNull.as("failed")
+      )
+
+      val issues: Seq[ValidationIssue] = dfWithConversion
+        .collect()
+        .map(x => (x.get(0), x.get(3)))
+        .toList
+        .collect {
+          case (description, true) => ValidationError(s"$description")
+      }
+      issues
+    }
+    catch {
+      case NonFatal(e) => Seq(ValidationError(e.getMessage))
+    }
+  }
+
+  private def defaultTimeZoneIssues(defaultTimeZone: Option[String]): Seq[ValidationIssue] = {
+    defaultTimeZone.filterNot(TimeZone.getAvailableIDs().contains(_)).map(tz =>
+      ValidationError(""""%s" is not a valid time zone designation""".format(tz))
+    ).toSeq
+  }
+
+  protected def patternAnalysisIssues(pattern: DateTimePattern,
+                                      defaultValue: Option[String],
+                                      defaultTimeZone: Option[String]): Seq[ValidationIssue]
+
+  protected def conversion(column: Column, pattern: DateTimePattern): Column
+}
+
+object FieldValidatorDateTime {
+
+  private val dayMilliSeconds = 24 * 60 * 60 * 1000
+  private val exampleDate = new Timestamp(System.currentTimeMillis)
+  private val epochStart = new Timestamp(0)
+  private val epochStartDayEnd = new Timestamp(dayMilliSeconds - 1)
+
+  private val spark: SparkSession = SparkSession.builder()
```

Comment:
Not sure about the creation timing. Need to check...

But anyway, I think it is way better to require validator users to provide a Spark session as a parameter (maybe implicit) rather than creating local sessions, even small local ones.

Reply:
We really shouldn't be creating additional SparkSessions.

Reply:
Hmm, hmm, hmm... 🤔 
It feels... cumbersome ... to require provision of `SparkSession` for the validator.
I was contemplating if to return to the use of `SimpleDateFormat`  (or `EnceladusDateTimeParser` which directly encapsulates it) when further investigation revealed to me Spark does use that for its casting. But I thought this way it's sure to work even if future Spark version would change the implementation.
This extra change, adding the SparkSession as an input parameter tilts the balance towards the `SimpleDateFormat`/`EnceladusDateTimeParser`, IMO.

Reply:
Let's not create local sessions.

Reply:
Ok, with that, let's close it 😃 . I will redo it with `SimpleDateFormat` (`EnceladusDateTimeParser`)